flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/JoinTest.scala

```diff
@@ -55,12 +55,13 @@ class JoinTest extends TableTestBase {
           unaryNode(
             "DataStreamCalc",
             streamTableNode(1),
-            term("select", "a", "b", "proctime")
+            term("select", "a", "b", "-(proctime, 3600000) AS -",
```

Comment:
Yes, I'm also curious why this plan would be generated. AFAIK, the `DataStreamWindowJoin` is very strict about the join predicates it accepts. It needs two range predicates with time indicator attributes on both sides. If the computation is pushed into the `DataStreamCalc` the time indicator attribute should be materialized and no longer be identified as a time indicator.